### PR TITLE
fix(rust-python): Join handles less frequently, fix backpressure bug

### PR DESCRIPTION
**Pull request by untitaker**
max_queue_depth does not work as it is implemented right now, but now
that we expose it as a parameter, we might as well use it to determine
how many handles are supposed to be in-flight.

The original theory was correct though: We are blocking too much in
check_for_results, and that's the culprit for our performance problems.

Now the next performance problem is the serialization overhead for
individual messages between processes, and we might have to implement
batching to fix that. But at the very least the processor can now
saturate 5 cores on my machine.

### CODE REVIEW

File: rust_snuba/src/strategies/python.rs [hunk 1]
```diff
@@ -81,82 +81,84 @@ impl PythonTransformStep {
...
+                tracing::error!(error, "Invalid message");
+            }
         }
     }
 }
 
 impl ProcessingStrategy<KafkaPayload> for PythonTransformStep {
     fn poll(&mut self) -> Result<Option<CommitRequest>, InvalidMessage> {
-        self.check_for_results();
+        if self.handles.len() > self.max_queue_depth {
```
Comment (lynnagara) [inline rust_snuba/src/strategies/python.rs:137]: Doesn't this mean we are *never* checking for results if we never hit max queue depth? How would that work for low volume topics/environments?

Comment (untitaker) [reply rust_snuba/src/strategies/python.rs:137]: ah you're right. i was relying on `join()` to eventually flush things out, but this doesn't work. how about a parameter called `max_queue_latency`?

Comment (lynnagara) [reply rust_snuba/src/strategies/python.rs:137]: I think that approach sounds good. Wdyt about not even making it a parameter and just hardcoding 1 second or something? I worry that having too many parameters for stuff makes it confusing/difficult for users to pick good values and it seems that a constant value here would be just fine at any volume really.

Comment (untitaker) [reply rust_snuba/src/strategies/python.rs]: done. i set it to 10 seconds because otherwise it has a lot of impact on throughput

**Review by lynnagara — APPROVED** [review]

**Pull request merged**